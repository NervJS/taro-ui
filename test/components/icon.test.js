import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtIcon from '../../.temp/components/icon/index'

describe('AtIcon Snap', () => {
  it('render initial AtIcon', () => {
    const component = renderToString(<AtIcon />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props value', () => {
    const component = renderToString(<AtIcon value='star' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props color', () => {
    const component = renderToString(<AtIcon color='#fff' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props size', () => {
    const component = renderToString(<AtIcon size='14' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props prefixClass', () => {
    const component = renderToString(
      <AtIcon prefixClass='prefixClass' value='star' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props customStyle', () => {
    const component = renderToString(<AtIcon customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIcon -- props className', () => {
    const component = renderToString(<AtIcon className='test' />)
    expect(component).toMatchSnapshot()
  })
})

describe('AtIcon Event', () => {
  it('AtIcon onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(<AtIcon onClick={onClick} />)
    const componentDom = findDOMNode(component, 'at-icon')

    Simulate.click(componentDom)
    expect(onClick).toBeCalled()
  })
})
