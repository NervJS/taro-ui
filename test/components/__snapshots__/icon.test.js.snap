// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtIcon Snap render AtIcon -- props className 1`] = `"<span style=\\"font-size:1.024rem;color:;\\" class=\\"taro-text at-icon test\\"></span>"`;

exports[`AtIcon Snap render AtIcon -- props color 1`] = `"<span style=\\"font-size:1.024rem;color:#fff;\\" class=\\"taro-text at-icon\\"></span>"`;

exports[`AtIcon Snap render AtIcon -- props customStyle 1`] = `"<span style=\\"font-size:1.024rem;color:;color:red;\\" class=\\"taro-text at-icon\\"></span>"`;

exports[`AtIcon Snap render AtIcon -- props prefixClass 1`] = `"<span style=\\"font-size:1.024rem;color:;\\" class=\\"taro-text prefixClass prefixClass-star\\"></span>"`;

exports[`AtIcon Snap render AtIcon -- props size 1`] = `"<span style=\\"font-size:0.5974rem;color:;\\" class=\\"taro-text at-icon\\"></span>"`;

exports[`AtIcon Snap render AtIcon -- props value 1`] = `"<span style=\\"font-size:1.024rem;color:;\\" class=\\"taro-text at-icon at-icon-star\\"></span>"`;

exports[`AtIcon Snap render initial AtIcon 1`] = `"<span style=\\"font-size:1.024rem;color:;\\" class=\\"taro-text at-icon\\"></span>"`;
